src/state.rs

```diff
@@ -1,46 +1,120 @@
-use chrono::{DateTime, Local};
+use anyhow::Result;
+use tui::widgets::ListState;
 
 use crate::tasks::Task;
 
 use std::fs::{self, File, OpenOptions};
 
-#[derive(Default)]
 pub struct State {
-    tasks: Vec<Task>,
+    pub text_input: String,
+    pub tasks: StatefulList<Task>,
+}
+
+impl Default for State {
+    fn default() -> Self {
+        State {
+            text_input: String::new(),
+            tasks: StatefulList::with_items(Vec::<Task>::new()),
+        }
+    }
 }
 
 impl State {
-    pub fn load_tasks(&mut self) -> std::io::Result<()> {
-        let file = open_file()?;
+    pub fn load_tasks(&mut self) -> Result<()> {
+        let file = open_file(FileMode::Load)?;
         let tasks: Vec<Task> = serde_json::from_reader(file)?;
-        self.tasks = tasks;
+        self.tasks = StatefulList::with_items(tasks);
         Ok(())
     }
 
-    pub fn save_tasks(&self) -> std::io::Result<()> {
-        let file = open_file()?;
-        serde_json::to_writer_pretty(file, &self.tasks)?;
+    pub fn save_tasks(&self) -> Result<()> {
+        let file = open_file(FileMode::Save)?;
```

Comment:
In rust default behaviour when reading/saving something to file is to NOT buffer io, so always remember to wrap a file in BufReader/BufWriter